### PR TITLE
Resolving issue with IByRefreshToken

### LINKED ISSUE #3736: [Bug] Incorrect HomeAccount details populated in AuthenticationResult of AcquireTokenByRefreshToken
Obtain Refresh token invoking oauth2 endpoint directly and using AcquireTokenByRefreshToken

1. Obtain ID token, access token, refresh token for the user using [https://login.microsoftonline.com/organizations/oauth2/v2.0/token endpoint using a Public Client App.

2. Use the above refresh token to call AcquireTokenByRefresh token method to obtain a new refresh token using MSAL. 
AcquireTokenByRefreshTokenParameterBuilder acquireTokenByRefreshTokenParameterBuilder = (app as IByRefreshToken).AcquireTokenByRefreshToken(
                scopes,
               refreshTokenHttp);
            AuthenticationResult authenticationResult = await acquireTokenByRefreshTokenParameterBuilder.ExecuteAsync(CancellationToken.None)
                .ConfigureAwait(false);

 Note that HomeAccountId details (Identifier, TenantId, ObjectId) in AuthenticationResult is completely incorrect.

Discussed this issue with @bgavrilMS @trwalke and they confirmed that the cause of the issue is that they don't send "client_info" = 1 with this request, so MSAL isn't able to cache the user properly.

### PR DESCRIPTION
**Pull request by trwalke**
Fixes #3736
[[Bug] Incorrect HomeAccount details populated in AuthenticationResult of AcquireTokenByRefreshToken](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3736)

**Changes proposed in this request**
Adding client info to IByRefreshToken request

**Testing**
Integration test

**Performance impact**
N/A

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs [hunk 1]
```diff
@@ -178,6 +179,44 @@ public async Task WithOnBeforeTokenRequest_TestAsync(Cloud cloud, TargetFramewor
...
+                .WithTestLogging()
+                .WithAuthority(labResponse.Lab.Authority, "organizations")
+                .BuildConcrete();
+
+            AuthenticationResult authResult = await msalPublicClient
+                .AcquireTokenByUsernamePassword(s_scopes, labResponse.User.Upn, labResponse.User.GetOrFetchPassword())
+                .ExecuteAsync(CancellationToken.None)
+                .ConfigureAwait(false);
+
+            var confidentialApp = ConfidentialClientApplicationBuilder
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs:199]: Hmm, you can build CCA without secret or cert?

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs [hunk 2]
```diff
@@ -178,6 +179,44 @@ public async Task WithOnBeforeTokenRequest_TestAsync(Cloud cloud, TargetFramewor
...
+                .WithAuthority(labResponse.Lab.Authority, labResponse.User.TenantId)
+                .WithTestLogging()
+                .BuildConcrete();
+
+            var rt = msalPublicClient.UserTokenCacheInternal.Accessor.GetAllRefreshTokens().FirstOrDefault();
+
+            // Act
+            authResult = await (confidentialApp as IByRefreshToken).AcquireTokenByRefreshToken(s_scopes, rt.Secret).ExecuteAsync().ConfigureAwait(false);
+
+            var account = authResult.Account;
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs:210]: The bug mentioned that `GetAccounts()` does not work. Can you use cca.GetAccount(id) nad cca.GetAccounts() ?

Comment (trwalke) [reply tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs:210]: @bgavrilMS  I dont see that mentioned in the issue but I will double check it.

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs (hunk 1 above)
Comment (trwalke) [reply tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs:199]: Dont think so, i think I did control z one too many times when cleaning up the code. Ill add back the credential.

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs (hunk 2 above)
Comment (bgavrilMS) [reply tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs:210]: Irrespective of the issue, please call `GetAccount(id)` and `GetAccounts()` to ensure that those APIs work.

**Review by bgavrilMS — APPROVED** [review]

**Review by pmaytak — APPROVED** [review]

**Review by SameerK-MSFT — APPROVED** [review]

**Pull request merged**